sheet.go

```diff
@@ -64,6 +64,19 @@ func (f *File) NewSheet(name string) int {
 	return f.GetSheetIndex(name)
 }
 
+// NewSheetWithRowNum provides the function to create a new sheet by given a worksheet and a given number of rows.
```

Comment:
This new API seems unnecessary, SetCellValue will preallocate cells for any new worksheets.

Reply:
Sometime will need this. Realloc cause huge GC pressure.
Example:
I need export a million row form db and write into excel, data is encrypt , need decrypt . 
slice often realloc we cause huge GC pressure 、copy spending

Reply:
I have encountered this situation before.
l remember alloc 1-3G mem. it's too Impressive.

Reply:
Sorry for my late reply. We can using streaming API [`NewStreamWriter`](https://pkg.go.dev/github.com/360EntSecGroup-Skylar/excelize/v2#File.NewStreamWriter) to generate huge data spreadsheet, by evaluation and follow the principle of least availability, I don't purpose to add this API. If you have any questions, let me know.